src/Pay/Adapter/Stripe.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+namespace Utopia\Pay\Adapter;
+
+use Utopia\Pay\Adapter;
+
+class Stripe extends Adapter
+{
+
+    private string $baseUrl = 'https://api.stripe.com/v1';
+    private string $secretKey;
+    private string $publishableKey;
+
+
+    public function __construct(string $publishableKey, string $secretKey, string $currency = 'USD')
+    {
+        $this->secretKey = $secretKey;
+        $this->publishableKey = $publishableKey;
+        $this->currency = $currency;
+    }
+
+    /**
+     * Get name of the payment gateway
+     */
+    public function getName(): string
+    {
+        return 'Stripe';
+    }
+
+    /**
+     * Make a purchase request
+     */
+    public function purchase(int $amount, string $customerId, string $cardId = null, array $additonalParams = []): array
+    {
+        $requestBody = [
+            'customer' => $customerId,
+            'amount' => $amount,
+            'currency' => $this->currency,
+        ];
+        $requestBody = array_merge($requestBody, $additonalParams);
+        if (!empty($cardId)) {
+            $requestBody['source'] = $cardId;
+        }
+        $res = $this->execute(self::METHOD_POST, '/charges', $requestBody);
+        return $res;
+    }
+
+    /**
+     * Refund payment
+     */
+    public function refund(string $paymentId, int $amount = null): array
+    {
+        $path = '/refunds';
+        $requestBody = ['charge' => $paymentId];
+        if ($amount != null) {
+            $requestBody['amount'] = $amount;
+        }
+        return $this->execute(self::METHOD_POST, $path, $requestBody);
+    }
+
+    /**
+     * Add a credit card for customer
+     */
+    public function createCard(string $customerId, string $cardId): array
+    {
+        $path = '/customers/' . $customerId . '/sources';
+        return $this->execute(self::METHOD_POST, $path, ['source' => $cardId]);
+    }
+
+    /**
+     * List cards
+     */
+    public function listCards(string $customerId): array
+    {
+        $path = '/customers/' . $customerId . '/sources';
+        return $this->execute(self::METHOD_GET, $path);
+    }
+
+    /**
+     * Update card
+     */
+    public function updateCard(string $customerId, string $cardId, string $name = null, int $expMonth = null, int $expYear = null, array  $billingDetails = null): array
```

Comment:
```suggestion
    public function updateCard(string $customerId, string $cardId, string $name = null, int $expMonth = null, int $expYear = null, array $billingDetails = null): array
```
There seems to be 2 spaces